src/iceberg/table.h

```diff
@@ -35,77 +36,96 @@ class ICEBERG_EXPORT Table {
+  /// \param[in] catalog The catalog that this table belongs to.
+  Table(TableIdentifier identifier, std::shared_ptr<TableMetadata> metadata,
+        std::string metadata_location, std::shared_ptr<FileIO> io,
+        std::shared_ptr<Catalog> catalog)
+      : identifier_(std::move(identifier)),
+        metadata_(std::move(metadata)),
+        metadata_location_(std::move(metadata_location)),
+        io_(std::move(io)),
+        catalog_(std::move(catalog)) {};
+
+  /// \brief Return the identifier of this table
+  const TableIdentifier& name() const { return identifier_; }
 
   /// \brief Returns the UUID of the table
-  virtual const std::string& uuid() const = 0;
+  const std::string& uuid() const;
 
-  /// \brief Refresh the current table metadata
-  virtual Status Refresh() = 0;
-
-  /// \brief Return the schema for this table
-  virtual const std::shared_ptr<Schema>& schema() const = 0;
+  /// \brief Return the schema for this table, return empty schema if not found
+  const std::shared_ptr<Schema>& schema() const;
 
   /// \brief Return a map of schema for this table
-  virtual const std::unordered_map<int32_t, std::shared_ptr<Schema>>& schemas() const = 0;
+  const std::unordered_map<int32_t, std::shared_ptr<Schema>>& schemas() const;
 
-  /// \brief Return the partition spec for this table
-  virtual const std::shared_ptr<PartitionSpec>& spec() const = 0;
+  /// \brief Return the partition spec for this table, return null if default spec is not
+  /// found
+  const std::shared_ptr<PartitionSpec>& spec() const;
 
   /// \brief Return a map of partition specs for this table
-  virtual const std::unordered_map<int32_t, std::shared_ptr<PartitionSpec>>& specs()
-      const = 0;
+  const std::unordered_map<int32_t, std::shared_ptr<PartitionSpec>>& specs() const;
 
-  /// \brief Return the sort order for this table
-  virtual const std::shared_ptr<SortOrder>& sort_order() const = 0;
+  /// \brief Return the sort order for this table, return null if default sort order is
+  /// not found
+  const std::shared_ptr<SortOrder>& sort_order() const;
 
   /// \brief Return a map of sort order IDs to sort orders for this table
-  virtual const std::unordered_map<int32_t, std::shared_ptr<SortOrder>>& sort_orders()
-      const = 0;
+  const std::unordered_map<int32_t, std::shared_ptr<SortOrder>>& sort_orders() const;
 
   /// \brief Return a map of string properties for this table
-  virtual const std::unordered_map<std::string, std::string>& properties() const = 0;
+  const std::unordered_map<std::string, std::string>& properties() const;
 
   /// \brief Return the table's base location
-  virtual const std::string& location() const = 0;
+  const std::string& location() const;
 
-  /// \brief Return the table's current snapshot
-  virtual const std::shared_ptr<Snapshot>& current_snapshot() const = 0;
+  /// \brief Return the table's current snapshot, return null if not found
+  std::shared_ptr<Snapshot> current_snapshot() const;
```

Comment:
```suggestion
  const std::shared_ptr<Snapshot>& current_snapshot() const;
```

Every table should have at least one snapshot. I think we can safely return a reference.